ArrayOperationTests/InsertAtTests.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Runtime.InteropServices;
+using Xunit;
+using ArrayOperations;
+
+namespace ArrayOperationTests
+{
+  public class InsertAtTests
+  {
+    [Fact]
+    public void InsertAt_Should_Throw_ArgumentException_When_Input_Array_Is_Empty()
```

Comment:
Why?  Why can't I insert into an empty array at index 0?

Reply:
Makes a lot of sense when you are asking, the user should be able to insert an item in an empty array.